pkg/cvo/updatepayload.go

```diff
@@ -82,28 +90,28 @@ func (r *payloadRetriever) RetrievePayload(ctx context.Context, update configv1.
 		return PayloadInfo{}, fmt.Errorf("no payload image has been specified and the contents of the payload cannot be retrieved")
 	}
 
-	var info PayloadInfo
-
 	// verify the provided payload
 	var releaseDigest string
 	if index := strings.LastIndex(update.Image, "@"); index != -1 {
 		releaseDigest = update.Image[index+1:]
 	}
-	verifyCtx := ctx
-
-	// if 'force' specified, ensure call to verify payload signature times out well before parent context
-	// to allow time to perform forced update
-	if update.Force {
-		timeout := time.Minute * 2
-		if deadline, deadlineSet := ctx.Deadline(); deadlineSet {
-			timeout = time.Until(deadline) / 2
-		}
-		klog.V(2).Infof("Forced update so reducing payload signature verification timeout to %s", timeout)
-		var cancel context.CancelFunc
-		verifyCtx, cancel = context.WithTimeout(ctx, timeout)
-		defer cancel()
+
+	var downloadCtx context.Context
+	var downloadCtxCancel context.CancelFunc
+	var verifyTimeout time.Duration
+
+	if deadline, ok := ctx.Deadline(); ok {
+		verifyTimeout = time.Until(deadline) / 2
+		downloadCtx = ctx
+	} else {
+		verifyTimeout = r.verifyTimeoutOnForce
+		downloadCtx, downloadCtxCancel = context.WithTimeout(ctx, r.verifyTimeoutOnForce+r.downloadTimeout)
```

Comment:
this looks like "if we don't use up our entire block of allocated time on verification, include any saved time in the allowed download window".  But I don't understand why having quicker verification would make us more generous in accepting slower downloads.  On the other hand, I expect that almost all clusters with faster verification will also have faster downloads, so there may be no clusters in the "fast verification, slow download" bucket to worry about.

Reply:
I'd ask the opposite, why shouldn't we be generous? Networks suck :) If we tolerate the whole `RetrievePayload` to take 4 minutes then we can occasionally soak a weird corner case this way and succeed.

But I'm not too strong on this, I can remove it if you worry about it making the code more complicated.

Reply:
When adding the godocs I have slightly refactored the code to better correspond to `RetrievePayload` carrying the contract: set just a single timeout on the whole method, and consider the division between verification/download an internal detail. 

It was hard to describe the semantics of `downloadTimeout` (it this plus sometimes a bit more but always bounded) and when things are hard to document it usually means something smells.